its/src/test/java/com/sonar/it/scanner/msbuild/sonarqube/CodeCoverageTest.java

```diff
@@ -0,0 +1,127 @@
+import com.sonar.it.scanner.msbuild.utils.VstsUtils;
+import com.sonar.orchestrator.build.BuildResult;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static com.sonar.it.scanner.msbuild.sonarqube.SonarQubeTestSuite.ORCHESTRATOR;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assumptions.assumeFalse;
+
+@ExtendWith(SonarQubeTestSuite.class)
+class CodeCoverageTest {
+  private static final String PROJECT_KEY = "code-coverage";
+  private static final String PROJECT_NAME = "CodeCoverage";
+
+  @TempDir
+  public Path basePath;
+
+  @BeforeEach
+  public void setUp() {
+    TestUtils.reset(ORCHESTRATOR);
+  }
+
+  @Test
+  void whenRunningOutsideAzureDevops_coverageIsNotImported() throws Exception {
+    // When running in AzureDevOps some of the environment variables are set and cannot be overwritten.
+    // Because of this, the coverage is always enabled.
+    assumeFalse(VstsUtils.isRunningUnderVsts());
+
+    var projectDir = TestUtils.projectDir(basePath, PROJECT_NAME);
+    var token = TestUtils.getNewToken(ORCHESTRATOR);
+
+    runBeginStep(projectDir, token, Collections.emptyList());
+    runTestsWithCoverage(projectDir);
+
+    var endStepResult = runEndStep(projectDir, token, Collections.emptyList());
+    assertThat(endStepResult.getLogs()).contains("'C# Tests Coverage Report Import' skipped because one of the required properties is missing");
+  }
+
+  @Test
+  void whenRunningOnAzureDevops_coverageIsImported() throws IOException {
+    var projectDir = TestUtils.projectDir(basePath, PROJECT_NAME);
+    var token = TestUtils.getNewToken(ORCHESTRATOR);
+
+    List<EnvironmentVariable> environmentVariables = VstsUtils.isRunningUnderVsts()
+      ? Collections.emptyList()
+      : // In order to simulate Azure-DevOps environment, the following variables are needed:
+        // TF_Build -> Must be set to true.
+        // BUILD_BUILDURI -> Must have a value. The value can be anything.
+        // AGENT_BUILDDIRECTORY -> The agent build directory; the tests results should be present in a child "TestResults" folder.
+        Arrays.asList(
+          new EnvironmentVariable("TF_Build", "true"),
+          new EnvironmentVariable("BUILD_BUILDURI", "fake-uri"),
+          new EnvironmentVariable("AGENT_BUILDDIRECTORY", projectDir.toString()));
+
+    runBeginStep(projectDir, token, environmentVariables);
+    runTestsWithCoverage(projectDir);
+
+    var endStepResult = runEndStep(projectDir, token, environmentVariables);
+    assertThat(endStepResult.getLogs()).contains("Coverage report conversion completed successfully.");
```

Comment:
After a rebase on the changes made in #1753, you can also assert the conversion itself. "Coverage report conversion completed successfully." does not indicate that an actual conversion occurred. You should additionally assert the `Converting coverage file '{0}' to '{1}'.` message to ensure the transformation from `.coverage` to `.coveragexml` occurred. You may have to enable `verbose=true` to see the message.